### DISCUSSION TITLE
Alexa Media Last Called

### DISCUSSION THREAD

**Discussion by ahahamyan** [alandtse/alexa_media_player#1357] [Q&A]
Hello, I have been using the component to have Alexa speak notifications successfully via HA. I have 10 echo devices throughout house, and in one case, I have a script that I want to execute via Alexa and have it speak back to (as a notification) the device it was spoken to. I created the script in HA and configured it to use notify.alexa_media_last_called instead of a specific echo (or group of echos). I have my HA enabled with Nabu Casa so, I created a routine in Alexa (which is invoked when I speak to Alexa)  which invokes the script (based on scene) in HA. All works perfectly when I am addressing a particular echo devices or group of echo devices in the HA script to be notified. However, what I really want is for the echo that was spoken to only respond back. So, when I used the notify.alexa_media_last_called the results are not consistent, sometimes the notification response is to the spoken echo and sometimes it is the previous echo spoken to. I also tried using the update_last_called service call, but it still did not make a difference. I added delays in between for a few seconds, but no avail. I also had the Alexa routine say something first before invoking scene, but that also did not work. Thanks so much for any help.

**Comment by ahahamyan** [comment]
Hello, just trying to understand if there are known issues with the notify.alexa_media_last_called . It appears that it is not working consistently and it appears that calling update_last_called service before notify  is not making a difference. Is there something I am doing wrong? Has this been tested and verified to work? I really want to be able to get the notification from the echo device I spoke to which was to invoke the script (as a scene). I use Home Assistant Cloud (Nabu Casa) and its skill. The Alexa Media Player component is very good and I love the integration. It appears this would just make it perfect. Thanks for any help in advance.

**Comment by ahahamyan** [comment]
I was able to get things working relatively consistently using approach below with delays of 1 second in between service calls

sequence:
  - delay:
      hours: 0
      minutes: 0
      seconds: 1
      milliseconds: 0
  - service: alexa_media.update_last_called
  - delay:
      hours: 0
      minutes: 0
      seconds: 1
      milliseconds: 0
-service: notify.alexa_media_last_called
data:
  message: (Script logic goes here)
  data:
    type: announce

However, I do get errors about 30 percent of the time with Unable to find service notify.alexa_media_last_called - When I retry the spoken command, it works. There is a reference to this issue at https://github.com/custom-components/alexa_media_player/issues/459 - while it appears the issue is closed, it still occurs.  

Thanks

**Reply by kenwiens** [reply]
Hi, did you make any progress here?  I would like to be able to ask Alexa my backyard temperature and get the value from one of my temperature sensors.  Sounds similar to what you were trying to do.  How did it go?

**Reply by ahahamyan** [reply]
Hi, it appears with the many updates since the last post, the following sequence of actions successively work perfectly on all my echo devices. Appears that there is no longer a need to have the delays in between calls. Thanks for the reminder as I checked this today and it worked perfectly every time. 

service: alexa_media.update_last_called
data: {}
enabled: true

service: notify.alexa_media_last_called
data:
  message: This is a test
  data:
    type: announce

**Comment by kenwiens** [comment]
Thanks.  I can make an automation that reports various temperatures around my house, and it plays on the last called Alexa.  It works perfectly when  I do it via "Run" in the automation developer section - playing on the last Alexa device that I had talked to.  However I am missing something pretty basic I guess - when I try to invoke this directly from Alexa via an Alexa routine or directly - (the automation seems to only be exposed as a "scene" to Alexa) - I can enable or disable it - but not "run" it.  The enable / disable flag in HA turns on or off when I turn the scene on or off in Alexa - but I can't get Alexa to actually invoke ("run") it.  I tried "Alexa open scene House climate report"  - and it gets me "Hmmm, I'm not sure"

If I create an Alexa routine that calls this scene (set as "power=on"), the appropriate alexa beeps as if it did something, enables or disables the scene (if the alexa setting was power on or off).  No attempt to actually "run" the yaml appears to be made and nothing shows up in the trace.

**Reply by kenwiens** [reply]
I just solved this problem. I was using an automation with no triggers (or conditions) - but you need to use a script instead. Honestly I don't think I saw that restriction anywhere in the reading I did.

**Reply by ahahamyan** [reply]
Yes, all of these Alexa based invocations I have are exposed as scripts in HA and they show up as scenes in Alexa when enabled through HomeAssistant Cloud which I subscribe to. I set up the routine in Alexa that invokes the scene (turns in on) - the response is asynchronous as it comes back from Alexa as an "announcement". This is as close to a request/response as we can get at the moment with Alexa integration. It works rather well. Glad you have it working.

**Comment by Morpheus2018** [comment]
media_player.home_fire_tab = "last_called: null", "last_called_timestamp: null", "last_called_summary: null" not work.

**Reply by narsaw** [reply]
Also having issue with alexa_media.update_last_called, stopped working about 2 weeks ago

**Reply by ahahamyan** [reply]
it looks like the service is not found - Unable to find service notify.alexa_media_last_called   - this is a very key feature, as this renders Alexa routines invoking HA scripts that provide response useless. 

[I created a new issue (https://github.com/custom-components/alexa_media_player/issues/2093)

**Reply by kenwiens** [reply]
This is being discussed in detail in issue #2089

https://github.com/custom-components/alexa_media_player/issues/2089
